db/types.go

```diff
@@ -116,9 +114,12 @@ func bz(s string) []byte {
 	return []byte(s)
 }
 
-// All DB funcs should panic on nil key.
-func panicNilKey(key []byte) {
-	if key == nil {
-		panic("nil key")
+// We defensively turn nil keys or values into []byte{} for
+// most operations.
+func nonNilBytes(bz []byte) []byte {
```

Comment:
Nice, this is more comforting to use than us panicking on every nil encounter :)